sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -330,4 +362,139 @@ object FileStreamSource {
 
     def size: Int = map.size()
   }
+
+  private[sql] class FileStreamSourceCleaner(
+      fileSystem: FileSystem,
+      sourcePath: Path,
+      baseArchiveFileSystem: Option[FileSystem],
+      baseArchivePath: Option[Path]) extends Logging {
+    require(baseArchiveFileSystem.isDefined == baseArchivePath.isDefined)
+
+    private val sourceGlobFilters: Seq[GlobFilter] = buildSourceGlobFilters(sourcePath)
+
+    private val sameFsSourceAndArchive: Boolean = {
```

Comment:
This should be an error, not something that causes the feature to be ignored. Just like providing a wrong value for `cleanSource`, or not providing the archive path, is an error.

Reply:
Oh OK. Sounds OK to fail the query and it might be even better than realizing later the configuration doesn't work since it was misconfigured or not supported. I'll make the change.